modules/account-lib/src/coin/hbar/transaction.ts

```diff
@@ -57,6 +57,27 @@ export class Transaction extends BaseTransaction {
     return toHex(this.encode(this._hederaTx, encoder));
   }
 
+  /**
+   * Sets this transaction payload
+   *
+   * @param rawTransaction
+   */
+  fromRawTransaction(rawTransaction: Uint8Array | string): void {
+    const buffer = typeof rawTransaction === 'string' ? toUint8Array(rawTransaction) : rawTransaction;
+    this.bodyBytes(buffer);
+    switch (this.txBody.data) {
+      case 'cryptoTransfer':
+        this.setTransactionType(TransactionType.Send);
+        break;
+      case 'cryptoCreateAccount':
+        this.setTransactionType(TransactionType.WalletInitialization);
+        break;
+      case 'tokenAssociate':
+        this.setTransactionType(TransactionType.AssociatedTokenAccountInitialization);
+        break;
```

Comment:
Is there any case in which there should be a `default` or should any of these be a `default` case?